### PR TITLE
printing: Remove curly braces from subscripted variables in python() and pycode()

### LINKED ISSUE #20762: pycode of latex expression with subscript leads to syntax error
Hey guys, while generating python code from a simple latex expression works

```python
expr = parse_latex(r'a \cdot b')
print(sympy.pycode(expr))
`>>> a*b
print(sympy.python(expr))
>>> a = Symbol('a')
>>> b = Symbol('b')
>>> e = a*b
```
for anything with a subscript the code generated by `pycode` or `.python` will lead to a syntax error due to the non-compliant variable name.

```python
expr = parse_latex(r'a_b \cdot b')
print(sympy.pycode(expr))
>>> a_{b}*b
print(sympy.python(expr))
>>> a_{b} = Symbol('a_{b}')  # this will generate a syntax error
>>> b = Symbol('b')
>>> e = a_{b}*b  # and this as well 
```

Cheers and thanks for the great work!

-------
versions
python = "3.9.1"
sympy = '1.7.1'
ANTLR runtime = "4.9.1"

### PR DESCRIPTION
**Pull request by AaronStiff**
#### References to other Issues or PRs

Fixes #20762 

#### Brief description of what is fixed or changed
Added handling for subscripted variables in `python` and `pycode`. Subscripted variables only become an issue when using a parsed latex expression, since latex wraps them in curly braces. `python` and `pycode` were creating variable names with curly braces in them, resulting in syntax errors. Added a simple check and modification in `python`, but redefined inherited `_print_Symbol` in `pycode`'s `PythonCodePrinter`.

#### Release Notes

* printing
    * Remove curly braces from subscripted variables in python() and pycode()

### CODE REVIEW

**Comment by AaronStiff** [comment]
Where should I put the test for this? It requires `antlr`, which doesn't get imported in `test_pycode.py` or `test_python.py`.

**Comment by oscargus** [comment]
> Where should I put the tests for this?

Either you check for it in the test, something like

```
antlr4 = import_module("antlr4")

# disable tests if antlr4-python*-runtime is not present
if not antlr4:
    disabled = True

```
so import antlr in the test and check if it is present, if not just return.

Or you may put the test in https://github.com/sympy/sympy/blob/master/sympy/parsing/tests/test_latex.py (where the code above is borrowed from).

File: sympy/printing/python.py [hunk 1]
```diff
@@ -51,7 +51,13 @@ def python(expr, **settings):
     # Returning found symbols and functions
     renamings = {}
     for symbolname in printer.symbols:
-        newsymbolname = symbolname
+        # Remove curly braces from subscripted variables
+        if '{' in symbolname:
+            newsymbolname = ''.join([s if s != '{' and s != '}' else '' for s in symbolname])
```
Comment (oscargus) [inline sympy/printing/python.py:56]: What about `newsymbolname = symbolname.replace('{', '').replace('}', '')`?

**Comment by AaronStiff** [comment]
Woah, I haven't touched sympy for half a year! :)

Would you mind taking over here?

**Comment by oscargus** [comment]
Sure, I can probably give it a go. Seems like it is just to sort out the test aspect (and possibly rewrite using replace).

**Comment by oscargus** [comment]
Couldn't figure out how to easily get your repo to my local computer so went for the "edit on GitHub"-approach. Not a brilliant idea in hindsight, but hopefully it will pass...

**Comment by oscarbenjamin** [comment]
> Couldn't figure out how to easily get your repo to my local computer so went for the "edit on GitHub"-approach. Not a brilliant idea in hindsight, but hopefully it will pass...

My approach to doing this is twofold. To check out a PR I have the following in `.git/config`:
```
[remote "upstream"]
  url = https://github.com/sympy/sympy.git
  fetch = +refs/heads/*:refs/remotes/upstream/*
  fetch = +refs/pull/*/head:refs/remotes/upstream/pr/*
  pushurl = no_push
```
With that I can do `git fetch upstream` and then any PR is available as `git checkout upsteram/pr/12345`.

Then if I want to make changes and push to the PR it's:
```
git checkout upstream/pr/20806
# make changes
git commit -am 'message'
git push git@github.com:AaronStiff/sympy.git HEAD:issues/20762-python-printing-subscripts
```
The username and branch name are taken from the header of the webpage in the PR on GitHub

**Pull request merged**